### PR TITLE
Fixed error for handling routes passed in constructor.

### PR DESCRIPTION
**Pull request by brapacz**
`EventRouter#initialize` expects a hash map of event names to _arrays_ of listeners. This PR relaxes this requirement by allowing either array or classes/lambdas.

### CODE REVIEW

File: spec/akasha/event_router_spec.rb [hunk 1]
```diff
@@ -7,16 +7,26 @@
...
+
+  describe '#connect' do
+    before do
+      subject.register_event_listener(:name_changed, Notifier)
+    end
+
+    include_examples 'registred actions'
+  end
+
+  context 'when passing single handler for event via constructor' do
```
Comment (bilus) [inline spec/akasha/event_router_spec.rb]: This should go to tests for `EventListenerBase`. All this test does is it verifies that the listener subscribes to repo.

File: spec/akasha/event_router_spec.rb [hunk 2]
```diff
@@ -7,16 +7,26 @@
...
+  shared_examples_for 'registred actions' do
     it "routes repo's events" do
       subject.connect!(repo)
       item.name = 'new name'
       expect_any_instance_of(Notifier).to receive(:on_name_changed).with('item-1', old_name: nil, new_name: 'new name')
       repo.save_aggregate(item)
     end
   end
+
+  describe '#connect' do
```
Comment (bilus) [inline spec/akasha/event_router_spec.rb]: `#connect!` -- my mistake :)

File: lib/akasha/event_router_base.rb [hunk 3]
```diff
@@ -3,7 +3,7 @@ module Akasha
   class EventRouterBase
     def initialize(routes = {})
       @routes = Hash.new { |hash, key| hash[key] = [] }
-      @routes.merge!(routes)
+      @routes.merge!(routes.map { |command, listeners| [command, Array(listeners)] }.to_h)
```
Comment (bilus) [inline lib/akasha/event_router_base.rb]: Would be good to add a comment, e.g. "Support both array of listeners and a single listener" or sth.

File: spec/akasha/event_router_spec.rb [hunk 4]
```diff
@@ -7,16 +7,26 @@
...
+    before do
+      subject.register_event_listener(:name_changed, Notifier)
```
Comment (bilus) [inline spec/akasha/event_router_spec.rb]: I think you might drop this before and use the constructor instead as it is the preferred way to register listeners.

**Review by bilus — CHANGES_REQUESTED** [review]

**Comment by bilus** [comment]
Also, update to CHANGELOG is missing @brapacz

File: lib/akasha/event_router_base.rb (hunk 3 above)
Comment (brapacz) [reply lib/akasha/event_router_base.rb]: I was thinking about it, but we already had this feature, but broken. I just fixed it.

File: spec/akasha/event_router_spec.rb (hunk 4 above)
Comment (brapacz) [reply spec/akasha/event_router_spec.rb]: Ok, I will drop it.

File: spec/akasha/event_router_spec.rb (hunk 1 above)
Comment (brapacz) [reply spec/akasha/event_router_spec.rb]: Ok, We should extract test suite for router's base and use it on every child class.

Comment (bilus) [reply spec/akasha/event_router_spec.rb]: Makes sense. 👍

File: lib/akasha/event_router_base.rb (hunk 3 above)
Comment (bilus) [reply lib/akasha/event_router_base.rb]: I mean a comment above this line so it's clear what it does. ❓

Comment (brapacz) [reply lib/akasha/event_router_base.rb]: I was thinking about changelog, sorry!

File: spec/akasha/async_event_router_spec.rb [hunk 5]
```diff
@@ -1,72 +1,79 @@
...
-    @thread = subject.connect!(repository)
-  end
-
-  after do
-    @thread.kill
-  end
-
-  context 'listener handling name_changed and count_changed' do
-    let(:listener) { FakeEventListener.new }
+  it_behaves_like 'EventRouter'
```
Comment (bilus) [inline spec/akasha/async_event_router_spec.rb]: `'event router``? `EventRouter` would suggest that you're referring to a class.

File: spec/akasha/event_router_base_spec.rb [hunk 6]
```diff
@@ -1,81 +1,3 @@
...
-
-    context 'with multiple valid listeners passed to constructor' do
-      let(:router) do
-        described_class.new(name_changed: [Notifier, ItemLogger])
-      end
-
-      include_examples 'routes registered events'
-    end
-  end
+  it_behaves_like 'EventRouter'
```
Comment (bilus) [inline spec/akasha/event_router_base_spec.rb]: `'event router'`

**Review by bilus — CHANGES_REQUESTED** [review]

**Review by bilus — APPROVED** [review]

**Pull request merged**